Docs: Plugin doc review for chunk 1-B (#67076)
grafana/grafana: docs/sources/developers/plugins/publish-a-plugin.md

type: docs
---

# Publish or update a plugin

You've just built your first plugin and [packaged it](package-a-plugin.md), and now you want to share it with the world.
In this guide, you'll learn how to share your plugin with others.

The best way to share your plugin with the world is to publish it in the [Grafana plugin catalog](/plugins).
By having your plugin published on Grafana.com, more users will be able to discover your plugin.

## Before you begin

Before you begin to publish a plugin, it needs to be built and packaged. 

> To build a plugin from source, Grafana creates a `dist` directory that contains the production build, or _plugin assets_, for your plugin.

Author: This PR is now closed. This issue is fixed in PR 67164.

> When the Grafana server starts, it recursively looks in the plugin directory for any directory that contains a `plugin.json` file, and then it tries to load the plugin assets in the same directory.

Reviewer: I don't see this sentence fitting in this context.

Author: This PR is now closed. This issue is fixed in PR 67164.

  If you're unable to make the source code publicly available, let us know in a comment on your plugin submission.
- **Tests:** We install your plugin on one of our Grafana instances to test it for basic use.
  For more advanced plugins, we may ask you to assist us in configuring a test environment for the plugin.

>   We will use the test environment whenever you submit a plugin update.

Reviewer 2: ```suggestion
  We'll use the test environment whenever you submit a plugin update.
```

     - Select **Multiple** if you'd like to submit separate plugin archives for each architecture.
       This can lead to faster downloads since users can select the specific architecture they want to install the plugin on.
   - **URL:** A URL that points to a ZIP archive of your packaged plugin.
   - **Source Code URL:** A URL that points to a public Git repository or ZIP archive of your complete plugin source code.
   - **MD5:** The MD5 hash of the plugin specified by the **URL**.
1. Click **Submit**.

## Frequently asked questions

**Do I need to submit a private plugin?**

- No. Please only submit plugins that you wish to make publicly available for the Grafana community.

**How long does it take to review my submission?**

- We're not able to give an estimate at this time, though we're constantly working on improving the time it takes to review a plugin.

**Can I decide a date when my plugin will be published?**

- No. We cannot guarantee specific publishing dates, as plugins are immediately published after a review based on our internal prioritization.

> **Can I see metrics of my plugin installs, downloads or usage?**

Reviewer 2: ```suggestion
**Can I see metrics of my plugin installs, downloads, or usage?**
```

Author: Done in PR 67070.

> - No. We don't offer this information at the moment to plugin authors.

Reviewer 2: ```suggestion
- No. We don't offer this information to plugin authors at the moment.
```

Author: Done in PR 67070.

**How can I update my plugin's catalog page?**

- The plugin's catalog page content is extracted from the plugin README file.
  To update the plugin's catalog page, submit an updated plugin with the new content included in the README file.

**Can I unlist my plugin from the Grafana plugin catalog in case of a bug?**

- In the event of a bug, unlisting the plugin from our catalog may be possible in exceptional cases, such as security concerns. However, we don't have control over the specific instances where the plugin is installed.

**Can I distribute my plugin somewhere else other than the Grafana plugin catalog?**

- The official method for distributing Grafana plugins is through our catalog. Alternative methods, such as installing private or development plugins on local Grafana instances, are available as per the guidelines provided in [this guide]({{< relref "../../administration/plugin-management#install-plugin-on-local-grafana" >}}).

**Can I still use Angular for my plugin?**

> - No. We will not accept any new plugin submissions written in angular. For more information, refer to our [Angular support deprecation documentation]({{< relref "../angular_deprecation" >}}).

Reviewer 2: ```suggestion
- No. We won't accept any new plugin submissions written in Angular. For more information, refer to our [Angular support deprecation documentation]({{< relref "../angular_deprecation" >}}).
```

Author: Done in PR 67070.
